File: src/features/settings/data/appSettings.ts
export type ThemeModeSetting = 'system' | 'light' | 'dark';

export interface AppSettings {
  readonly localeOverride: string | null;
  readonly themeMode: ThemeModeSetting;
}

export const defaultSettings = (): AppSettings => ({ localeOverride: null, themeMode: 'system' });

File: lib/features/settings/data/app_settings.dart
import 'package:flutter/material.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'app_settings.freezed.dart';

@freezed
abstract class AppSettings with _$AppSettings {
  const factory AppSettings({
    required Locale? localeOverride,
    required ThemeMode themeMode,
  }) = _AppSettings;
}
